solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -368,4 +341,77 @@ public static Map<String, Object> postProcessCollectionJSON(Map<String, Object>
     collection.put("health", Health.combine(healthStates).toString());
     return collection;
   }
+
+  private class SolrCollectionProperiesIterator implements Iterator<NamedList<Object>> {
```

Comment:
I was hoping we wouldn't need a custom Iterator.  We do need a method that takes in DocCollection (and some other context that is the same across collections) and returns a NamedList.  With such a method, we can call it via streamOfDocCollection.map(collState -> theMethod(collState, routeKey, liveNodes, etc.).iterator()

Reply:
Ah yeah don't know why I didn't do this first. Changed it appropriately.